AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift

```diff
@@ -0,0 +1,236 @@
+    ///
+    /// - Parameters:
+    ///   - logger: Custom logger to log events.
+    public init(logger: Logger) {
+        self.logger = logger
+        segmentationProcessor = SegmentationProcessor(logger: self.logger)
+    }
+
+    /// Creates the alpha mask [0-255] of the foreground image using `SegmentationProcessor`.
+    ///
+    /// - Parameters:
+    ///   - inputFrameCG: Input CGImage frame to produce the foreground image.
+    ///   - inputFrameCI: Input CIImage frame to produce the foreground image.
+    ///
+    /// - Returns: Alpha mask CGImage of the foreground.
+    public func createForegroundAlphaMask(inputFrameCG: CGImage,
+                                          inputFrameCI: CIImage) -> CIImage? {
+        #if canImport(AmazonChimeSDKMachineLearning)
+        // Number of the input image color space channels.
+        let imageChannels = inputFrameCG.bitsPerPixel / inputFrameCG.bitsPerComponent
+
+        // Update the buffer pool dimensions if the new frame does not match the previous frame dimensions.
+        if bufferPool == nil || inputFrameCG.width != bufferPoolWidth || inputFrameCG.height != bufferPoolHeight {
+
+            updateBufferPool(newWidth: inputFrameCG.width, newHeight: inputFrameCG.height)
+            do {
+                // Initialize the segmentationProcessor if it has not been initialized.
+                try segmentationProcessor.initialize(height: segmentationProcessorHeight,
+                                                     width: segmentationProcessorWidth,
+                                                     channels: Int32(imageChannels))
+            } catch ResourceError.notFound {
+                logger.error(msg: "Unable to find selfie segmentation model." +
+                             "See `Update Project File` section in README for more information on " +
+                             "how to import `selfie_segmentation_landscape.tflite` as a bundle resource to your project.")
+                return nil
+            } catch {
+                logger.error(msg: "Unexpected error: \(error).")
+            }
+        }
+
+        // Calculate the scale and aspect ratio factor to downscale.
+        let downSampleScale = Double(segmentationProcessorHeight) / Double(inputFrameCG.height)
+        let downSampleAspectRatio = Double((Double(inputFrameCG.height)/Double(inputFrameCG.width)) /
+                                           (Double(segmentationProcessorHeight)/Double(segmentationProcessorWidth)))
+        // Down sample the image.
+
+        guard let downSampleImage: CIImage = resizeImage(image: inputFrameCI,
+                                                         scale: downSampleScale,
+                                                         aspectRatio: downSampleAspectRatio)
+        else {
+            return nil
+        }
+
+        // Check if segmentation model has loaded.
+        if segmentationProcessor.getModelState() != CwtModelState.LOADED {
```

Comment:
shouldn't we have this if on the top before `imageChannels`?

Reply:
We need to check if the model was loaded after the initialization on line 74